pkg/lib/machine-dialogs.js

```diff
@@ -640,6 +641,28 @@ function ChangeAuth(dialog) {
             }
         }
 
+        if (key_locked_div && error_options &&
+            error_options.error && error_options.error.startsWith("locked identity")) {
+            console.log("Locked identitiy: ", error_options);
+            const identity_path = error_options.error.split(": ")[1];
+            key_locked_div.find(".locked-identity").text(identity_path);
+            key_locked_div.find(".btn-primary").on("click", () => {
+                key_locked_div.find(".btn-primary").attr("disabled", true);
+                keys.load(identity_path, key_locked_div.find(".locked-identity-password").val())
+                        .done(() => dialog.render_error([]))
```

Comment:
Passing `[]` here looks a little weird -- in the case of an exception it passes an ex object, not an array. This would look clearer with `null`? This gets passed to `cockpit.message()` as well, which expects an object, not an array.

Also, could this please use `.then()` and `.catch()`, instead of the deprecated jquery promise handlers?

Reply:
This is a bit of a hack to reuse the error reporting within the dialog. Normally it waits until you click accept to clear the error. Maybe i should just add an explicit "clear_errors" function? Passing an empty object or null doesn't work, results in an error message being present.